src/ui/VirtualFileRenderer/VirtualFileRenderer.tsx

```diff
@@ -0,0 +1,323 @@
+import { useWindowVirtualizer } from '@tanstack/react-virtual'
```

Comment:
I've found that leaving a comment on how the virtualization is done (particularly the textarea trick and how it enables native cmd+f) is something worth writing about here. It will help future readers understand how it work and learn something without having to look at the details of the implementation

Reply:
Will do 🫡